### PR TITLE
[fli-iam#1846] Correct order by numeric value in lists

### PR DESCRIPTION
**Pull request by youennmerel**
In study list :
- Sorting by "storage volume" is now correct
- Sorting on "storage volume" is disabled while fetching
- ~Storage volumes are fetched only for studies loaded in the current page~ (not possible due to sorting)

Correct https://github.com/fli-iam/shanoir-ng/issues/1868

How to test :
- Display study list as admin (so all the studies will be loaded, it will take some time)
- Order by "storage volume", asc & desc (must be disabled while fetching)
- Display study details

### CODE REVIEW

File: shanoir-ng-front/src/app/studies/study/study.component.ts [hunk 1]
```diff
@@ -118,6 +118,9 @@ export class StudyComponent extends EntityComponent<Study> {
     }
 
     initView(): Promise<void> {
+
+        console.log(this.breadcrumbsService.currentStep.entity);
```
Comment (jcomedouteau) [inline shanoir-ng-front/src/app/studies/study/study.component.ts:122]: Remove debug logs ?

File: shanoir-ng-front/src/app/studies/study-list/study-list.component.ts [hunk 2]
```diff
@@ -96,33 +99,41 @@ export class StudyListComponent extends BrowserPaginEntityListComponent<Study> {
...
-                                sizesByLabel.set(DatasetExpressionFormat.getLabel(sizeByFormat.format), sizeByFormat.size);
+            this.fetchStorageVolumes(studies);
+        });
+        return earlyResult;
+    }
+
+    private fetchStorageVolumes(studies: Study[] | AccessRequest[]) {
+        let pageSize = Number(this.table.maxResults);
+        let promises = [];
+        for (let i = 0; i < studies.length; i += pageSize) {
```
Comment (jcomedouteau) [inline shanoir-ng-front/src/app/studies/study-list/study-list.component.ts:110]: Why do we do it page by page ? Is it too long otherwise ?

Comment (jcomedouteau) [reply shanoir-ng-front/src/app/studies/study-list/study-list.component.ts:110]: It is quite weird that we can start to sort by size wereas all study size are not loaded yet -> The display can be then misleading

Comment (michaelkain) [reply shanoir-ng-front/src/app/studies/study-list/study-list.component.ts:110]: Hi! I agree with Jean-Come. Users will find our tool strange, when we propose a filter on size, but only consider the current page.

**Review by jcomedouteau — APPROVED** [review]

**Pull request merged**